lib/at_cmd_parser/at_params.c

```diff
@@ -195,25 +200,54 @@ int at_params_string_put(const struct at_param_list *list, size_t index,
 		return -EINVAL;
 	}
 
-	char *param_value = k_malloc(str_len + 1);
+	char *param_value = (char *)k_malloc(str_len + 1);
 
 	if (param_value == NULL) {
 		return -ENOMEM;
 	}
 
 	memcpy(param_value, str, str_len);
-	param_value[str_len] = '\0';
 
 	at_param_clear(param);
+	param->size = str_len;
 	param->type = AT_PARAM_TYPE_STRING;
-	param->value.str_val =	param_value;
+	param->value.str_val = param_value;
+
+	return 0;
+}
+
+int at_params_array_put(const struct at_param_list *list, size_t index,
+			 const u32_t *array, size_t array_len)
+{
+	if (list == NULL || list->params == NULL || array == NULL) {
+		return -EINVAL;
+	}
+
+	struct at_param *param = at_params_get(list, index);
+
+	if (param == NULL) {
+		return -EINVAL;
+	}
+
+	u32_t *param_value = (u32_t *)k_malloc(array_len);
```

Comment:
You should update the `at_param_clear` function and free the memory allocated for param of type `AT_PARAM_TYPE_ARRAY`. Currently, we have a memory leak.

Reply:
And it should be clarified in the documentation that `array_len` is the size of the array in bytes, not number of elements. Possibly could rename to `array_size`.

Reply:
Thanks, good catch.